src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -190,6 +198,66 @@ public Response updateByDatasetUpdate(
     }
   }
 
+  /**
+   * This endpoint updates the dataset.
+   */
+  @PATCH
+  @Consumes({MediaType.APPLICATION_JSON})
+  @Produces({MediaType.APPLICATION_JSON})
+  @Path("/{datasetId}")
+  @RolesAllowed({ADMIN, CHAIRPERSON, DATASUBMITTER})
+  public Response patchByDatasetUpdate(@Auth AuthUser authUser,
+      @PathParam("datasetId") Integer datasetId, String json) {
+    try {
+      if (json == null || json.isEmpty()) {
+        throw new BadRequestException("Dataset Patch is required");
+      }
+      Gson gson = GsonUtil.getInstance();
+      DatasetPatch patch;
+      try {
+        patch = gson.fromJson(json, DatasetPatch.class);
+      } catch (Exception e) {
+        throw new BadRequestException("Unable to parse dataset patch: " + json);
+      }
+      Dataset existingDataset = datasetService.findDatasetById(datasetId);
+      if (existingDataset == null) {
+        throw new NotFoundException("Could not find the dataset with id: " + datasetId);
+      }
+      // Check permissions for non-admin roles.
+      User user = userService.findUserByEmail(authUser.getEmail());
+      if (!user.hasUserRole(UserRoles.ADMIN)) {
+        if (!isCreatorOrCustodian(user, existingDataset)) {
+          throw new ForbiddenException("User does not have permission to update this dataset");
+        }
+      }
+      if (!patch.isPatchable(existingDataset)) {
+        return Response.notModified().entity(existingDataset).build();
+      }
+      // Validate DatasetPatch values
+      List<String> existingNames = datasetService.findAllDatasetNames();
+      if (patch.name() != null && !patch.name().equals(existingDataset.getName())
+          && existingNames.contains(patch.name())) {
+        return Response.status(HttpStatusCodes.STATUS_CODE_BAD_REQUEST)
+            .entity(new Error(
+                "The new name for this dataset already exists: " + patch.name(),
+                HttpStatusCodes.STATUS_CODE_BAD_REQUEST))
+            .build();
+      }
+      if (!patch.validateProperties()) {
+        return Response.status(HttpStatusCodes.STATUS_CODE_BAD_REQUEST)
+            .entity(new Error(
+                "Properties are invalid",
+                HttpStatusCodes.STATUS_CODE_BAD_REQUEST))
+            .build();
```

Comment:
From what I read about `WebApplicationException`, you could throw here to avoid making a response. This should have the same effect:
```suggestion
        throw new BadRequestException("Properties are invalid");
```